py/client/pydeephaven/updateby.py

```diff
@@ -242,3 +301,1072 @@ def ema_time_decay(ts_col: str, decay_time: int, cols: Union[str, List[str]],
+
+    Returns:
+        an UpdateByOperation
+
+    Raises:
+        DHError
+     """
+    try:
+        window_scale = _GrpcUpdateByWindowScale(time=_GrpcUpdateByWindowTime(column=ts_col, period_nanos=decay_time))
+        ub_ems = _GrpcUpdateByEms(options=op_control.make_grpc_message() if op_control else None,
+                                  window_scale=window_scale)
+        ub_spec = _GrpcUpdateBySpec(ems=ub_ems)
+        ub_column = _GrpcUpdateByColumn(spec=ub_spec, match_pairs=to_list(cols))
+        return UpdateByOperation(ub_column=ub_column)
+    except Exception as e:
+        raise DHError("failed to create a time-decay EMS UpdateByOperation.") from e
+
+
+def emmin_tick(decay_ticks: float, cols: Union[str, List[str]],
+               op_control: OperationControl = None) -> UpdateByOperation:
+    """Creates an EM Min (exponential moving minimum) UpdateByOperation for the supplied column names, using ticks as
+    the decay unit.
+
+    The formula used is
+        a = e^(-1 / decay_ticks)
+        em_val_next = min(a * em_val_last, value)
+
+    Args:
+        decay_ticks (float): the decay rate in ticks
+        cols (Union[str, List[str]]): the column(s) to be operated on, can include expressions to rename the output,
+            i.e. "new_col = col"; when empty, update_by perform the operation on all columns.
+        op_control (OperationControl): defines how special cases should behave, when None, the default OperationControl
+            settings as specified in :meth:`~OperationControl.__init__` will be used
+
+    Returns:
+        an UpdateByOperation
+
+    Raises:
+        DHError
+    """
+    try:
+        window_scale = _GrpcUpdateByWindowScale(ticks=_GrpcUpdateByWindowTicks(ticks=decay_ticks))
+        ub_emmin = _GrpcUpdateByEmMin(options=op_control.make_grpc_message() if op_control else None,
+                                  window_scale=window_scale)
+        ub_spec = _GrpcUpdateBySpec(em_min=ub_emmin)
+        ub_column = _GrpcUpdateByColumn(spec=ub_spec, match_pairs=to_list(cols))
+        return UpdateByOperation(ub_column=ub_column)
+    except Exception as e:
+        raise DHError("failed to create a tick-decay EMS UpdateByOperation.") from e
+
+
+def emmin_time(ts_col: str, decay_time: int, cols: Union[str, List[str]],
+               op_control: OperationControl = None) -> UpdateByOperation:
+    """Creates an EM Min (exponential moving minimum) UpdateByOperation for the supplied column names, using time as the
+    decay unit.
+
+    The formula used is
+        a = e^(-dt / decay_time)
+        em_val_next = min(a * em_val_last, value)
+
+     Args:
+        ts_col (str): the column in the source table to use for timestamps
+        decay_time (int): the decay rate, in nanoseconds
+        cols (Union[str, List[str]]): the column(s) to be operated on, can include expressions to rename the output,
+            i.e. "new_col = col"; when empty, update_by perform the operation on all columns.
```

Comment:
```suggestion
            i.e. "new_col = col"; when empty, update_by performs the operation on all columns.
```
We are inconsistent on our `cols` docstring.  Sometimes we list the operation that is being performed.  Sometimes we incorrectly list the operation because of cut and paste.  Other times, like this, we have a generic string.  Maybe use the generic string everywhere to avoid cut and paste bugs?

If this is getting pasted in many places, the `perform` ->`performs` change needs to happen everywhere.